umrover/mrover-ros
# Add ESW motors code

## NOTE
THIS PR WILL SOON CLOSE ONCE QUINTIN FILTERS THE HISTORY ON THE OTHER BRANCHES (unless he does it on my branches too)

This is NOT for the motors on used in the drive system (THIS IS NOT THE CODE THAT CONTROLS THE ODRIVES/WHEELS). This code controls the motors used on the robotic arm, the science acquisition system (this year, the science arm) and all other brushed motors such as the mast gimbal motor.

## Summary
Addresses Issue: #34 

Move all of Rosie's '22 arm/mast gimbal/other motor code into ROS. 

### Did you add documentation to the wiki?
Yes (Documentation in ESW wiki should be sufficient for a brief overview).

## How was this code tested? 
It was not. (That's why it's still in draft)

### Did you test this in sim? 
Not applicable.

### Did you test this on the rover?
Not yet.

### Did you add unit tests? 
Yes (to some extent). The unit tests are used mainly to test the arm itself though rather than the code.

## TODO
- think about how we want to even run tests
- start working on test file 
- config file is untested
- syntax is likely very wrong
- change hardware stuff once year comes around and systems are fleshed out
- retest everything
- fix up test file
- coding style is all over the place -- functions should have verbs.
- functions should not have underscores
- just follow this guide https://google.github.io/styleguide/cppguide.html
- make sure we have const everywhere applicable
- avoid #define whenever possible (and that means, essentially always). Use C++ language
features (enums, constexpr variables, template functions, etc.) instead
